converters/lance-image-dataset-converter/convert-any-image-dataset-to-lance.py

```diff
@@ -0,0 +1,106 @@
+import os
+import argparse
+import pandas as pd
+import pyarrow as pa
+import lance
+import time
+from tqdm import tqdm
+
+def process_images(images_folder, data_type):
+    # Define schema for RecordBatch
+    schema = pa.schema([('image', pa.binary()), 
+                        ('filename', pa.string()), 
+                        ('category', pa.string()), 
+                        ('data_type', pa.string())])
+
+    # Iterate over the categories within each data type
+    category_folder = os.path.join(images_folder, data_type)
+    for category in os.listdir(category_folder):
+        category_folder = os.path.join(images_folder, data_type, category)
+        
+        # Iterate over the images within each category
+        for filename in tqdm(os.listdir(category_folder), desc=f"Processing {data_type} - {category}"):
+            # Construct the full path to the image
+            image_path = os.path.join(category_folder, filename)
+
+            # Read and convert the image to a binary format
+            with open(image_path, 'rb') as f:
+                binary_data = f.read()
+
+            image_array = pa.array([binary_data], type=pa.binary())
+            filename_array = pa.array([filename], type=pa.string())
+            category_array = pa.array([category], type=pa.string())
+            data_type_array = pa.array([data_type], type=pa.string())
+
+            # Yield RecordBatch for each image
+            yield pa.RecordBatch.from_arrays(
+                [image_array, filename_array, category_array, data_type_array],
+                schema=schema
+            )
+
+# Function to write PyArrow Table to Lance dataset
+def write_to_lance(images_folder):
+    # Create an empty RecordBatchIterator
+    schema = pa.schema([
+        pa.field("image", pa.binary()),
+        pa.field("filename", pa.string()),
+        pa.field("category", pa.string()),
+        pa.field("data_type", pa.string())
+    ])
+
+    for data_type in ['train', 'test', 'val']:
+        lance_file_path = os.path.join(images_folder, f"mini_imagenet_{data_type}.lance")
```

Comment:
Since our script will convert any image dataset, it would be better to extract the name of the dataset and use that here, instead of "mini_imagenet_{data_type}.lance"